src/parser.rs

```diff
@@ -523,12 +523,21 @@ impl Parser {
         // Following the string literal is a qualifier which indicates the units
         // of the duration specified in the string literal.
         //
-        // Note that PostgreSQL allows omitting the qualifier, but we currently
-        // require at least the leading field, in accordance with the ANSI spec.
-        let leading_field = self.parse_date_time_field()?;
+        // Note that PostgreSQL allows omitting the qualifier, so we provide
+        // this more general implemenation.
+        let leading_field = match self.peek_token() {
+            Some(Token::Word(kw))
+                if ["YEAR", "MONTH", "DAY", "HOUR", "MINUTE", "SECOND"]
+                    .iter()
+                    .any(|d| kw.keyword == *d) =>
```

Comment:
I'm not entirely sure about this approach, but it does seem to work. I certainly would appreciate your thoughts. 🙂

Reply:
I guess this will do for now, seems like a good place to use the `maybe_parse` helper from a sibling PR, as well as `parse_set` https://github.com/andygrove/sqlparser-rs/blob/d842f495db6936f64b2738178a431f0ebd6c4e07/src/parser.rs#L1769-L1770